### PR TITLE
Make logger compatible with monolog (PSR-3)

### PR DESCRIPTION
**Pull request by S-anasol**
[PSR-3](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-3-logger-interface.md)

### CODE REVIEW

File: lib/Connection.php [hunk 1]
```diff
@@ -314,8 +314,8 @@ public function query($sql, &$values=array())
 	{
 		if ($this->logging)
 		{
-			$this->logger->log($sql);
-			if ( $values ) $this->logger->log($values);
+			$this->logger->debug($sql);
+			if ( $values ) $this->logger->debug("Values:", $values);
```
Comment (Rican7) [inline lib/Connection.php]: This change isn't enough. [The `Config` class does reflection to verify that the passed instance actually implements a public `log` method.](https://github.com/jpfuentes2/php-activerecord/blob/891ff0d5d4f5423583a31538f48fddd75321c577/lib/Config.php#L247-L248)

Comment (S-anasol) [inline lib/Connection.php]: something like this? :dancers:

**Comment by Rican7** [comment]
I'm going to have to give this PR a 👎 

This change needlessly breaks backwards-compatibility with existing log setups. While PSR-3 would be ideal to support, it doesn't make sense to break BC in `master` just to do it. We'd have to make it a new major release for semver.

Even still, though, it's mostly unnecessary as it's quite trivial to make [an adapter](https://en.wikipedia.org/wiki/Adapter_pattern) for PSR-3 and PHP-ActiveRecord:

```php
<?php

use Psr\Log\LogLevel;
use Psr\Log\LoggerInterface;

/**
 * A PSR-3 logger adapter for PHP-ActiveRecord
 */
class ActiveRecordLogger
{
    /**
     * The default PSR-3 compatible log level for log entries.
     */
    const DEFAULT_LEVEL = LogLevel::DEBUG;

    /**
     * A PSR-3 logger to delegate logging to.
     *
     * @type LoggerInterface
     */
    private $logger;

    /**
     * The PSR-3 compatible log level to use for log entries.
     *
     * @see LogLevel
     * @type int
     */
    private $level;


    /**
     * Constructor
     *
     * @param LoggerInterface $logger A PSR-3 logger to delegate logging to.
     * @param int $level The PSR-3 compatible log level to use for log entries.
     */
    public function __construct(LoggerInterface $logger, $level = self::DEFAULT_LEVEL)
    {
        $this->logger = $logger;
        $this->level = $level ?: self::DEFAULT_LEVEL;
    }

    /**
     * Logs a message to the delegate logger.
     *
     * This method is the PHP-ActiveRecord required log method.
     *
     * @param string $message The message to log.
     * @return void
     */
    public function log($message)
    {
        $this->logger->log($this->level, $message);
    }
}
```

It's a total of 1-line of functional code, which is just a delegate call.

**Comment by jpfuentes2** [comment]
Thanks for your contribution, @S-anasol, but I agree with @Rican7 that you can fix this in your own codebase simply and that we'd need a proper release to change this. Closing since this is not our current focus.

**Comment by S-anasol** [comment]
Ofc changed locally, PR just for reference :+1:

**Pull request closed without merging**